pvlib/modelchain.py

```diff
@@ -1796,3 +1804,8 @@ def _tuple_from_dfs(dfs, name):
         return tuple(df[name] for df in dfs)
     else:
         return dfs[name]
+
+
+@_unwrap_single_value
+def _to_tuple(x):
+    return tuple(x)
```

Comment:
We have to be explicit about the types we are converting _from_ here, otherwise we could see unexpected behavior (e.g. if you pass a DataFrame to `tuple()` you will end up with a tuple of column names).
```suggestion
def _to_tuple(x):
    if not isinstance(x, (list, tuple)):
        return x
    return tuple(x)
```
This lets us drop the decorator entirely.

Reply:
I was not confident that simple return tuple(x) would cover all the cases.